docs/tools/node_maintain.md

```diff
@@ -0,0 +1,85 @@
+# node_maintain
+
+## Overview
+
+Graceful decommissioning could minimize the impact on existing job when removing unhealthy nodes. 
+This is a tool to help you achieve it in OpenPAI. 
+
+In OpenPAI, an unhealthy node-list records the nodes to be decommissioned. 
+Components could sync the list and decommission them as needed.
+
+**NOTES: For now, only YARN nodes are decommissioned, HDFS doesn't.**
+
+## Commands
+
+Currently, node-list update and component refresh are both manual actions. 
+This tool provides corresponding command for them. 
+The only mandatory argument for all commands is `master_ip`. Usually, components share the same master ip,
+in different case(i.e. load-balance), you could overwrite `master_ip` separately by `--kubernetes-ip` and `--yarn-ip`
+
+So, a common invoking looks like:
+```bash
+python node_maintain.py {master_ip} sub-command sub-command-arguments
+```
+
+Noticed that execution path should be `{PAI_ROOT_DIR}/src/tools`.
+
+### node-list sub-command
+
+`node-list` command could get and edit unhealthy node-list, 
+notice that the change will not trigger the real refresh behavior.
+
+* `node-list get`: print current unhealthy node-list
+* `node-list add nodes`: add nodes to current node-list
+* `node-list remove nodes`: remove nodes from current node-list
+* `node-list update nodes`: overwrite current node-list with nodes
+
+
+### refresh sub-command
+
+Enforce the node-list, it's a blocking command and won't exit until all nodes in node-list are decommissioned.
+
+
+## Scenarios
+
+For convenience, in this section, we assume you have a cluster without load-balance, as below:
+```
+10.0.0.10 master
+10.0.0.1 work-1
+10.0.0.2 work-2
+10.0.0.3 work-3
+```
+
+### Decommission nodes
```

Comment:
>Decommission [](start = 4, length = 12)

Please add some doc on how to recommission a node

Reply:
I think this example is enough?
https://github.com/Microsoft/pai/blob/859f291c0fc1babe54e643d54c4859f057ffafc5/docs/tools/node_maintain.md#recommission-nodes